openhands/runtime/utils/runtime_templates/Dockerfile.j2

```diff
@@ -70,16 +63,26 @@ RUN \
     /openhands/micromamba/bin/micromamba run -n openhands poetry run playwright install --with-deps chromium && \
     # Set environment variables
     echo "OH_INTERPRETER_PATH=$(/openhands/micromamba/bin/micromamba run -n openhands poetry run python -c "import sys; print(sys.executable)")" >> /etc/environment && \
-    # Install extra dependencies if specified
-    {{ extra_deps }} {% if extra_deps %} && {% endif %} \
     # Clear caches
     /openhands/micromamba/bin/micromamba run -n openhands poetry cache clear --all . && \
     # Set permissions
-    {% if not skip_init %}chmod -R g+rws /openhands/poetry && {% endif %} \
+    chmod -R g+rws /openhands/poetry && \
     mkdir -p /openhands/workspace && chmod -R g+rws,o+rw /openhands/workspace && \
     # Clean up
     apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
     /openhands/micromamba/bin/micromamba clean --all
+
+# ================================================================
+# END: Build Runtime Image from Scratch
 # ================================================================
-# END: Copy Project and Install/Update Dependencies
+{% endif %}
+
 # ================================================================
+# Copy Project source files
+# ================================================================
+RUN if [ -d /openhands/code/openhands ]; then rm -rf /openhands/code/openhands; fi
+COPY ./code/pyproject.toml ./code/poetry.lock /openhands/code
+COPY ./code/openhands /openhands/code/openhands
```

Comment:
I think we need to re-run the poetry install here when the new code gets copied in - so the new dependencies will be correctly installed?

Reply:
If the pyproject / poetry.lock is exactly the same, the poetry install should finish really quick

Reply:
I updated this code - see the notes about a 2 part hash.